projects/adrv9009_zu11eg_som/carrier_bd.tcl

```diff
@@ -1,2 +1,88 @@
 
 add_files -fileset constrs_1 -norecurse ./carrier_constr.xdc
+
+create_bd_port -dir O -type clk i2s_mclk
+create_bd_intf_port -mode Master -vlnv analog.com:interface:i2s_rtl:1.0 i2s
+
+# 12.288MHz clk
+ad_ip_instance axi_clkgen sys_audio_clkgen
+ad_ip_parameter sys_audio_clkgen CONFIG.ID 6
+ad_ip_parameter sys_audio_clkgen CONFIG.CLKIN_PERIOD 10
+ad_ip_parameter sys_audio_clkgen CONFIG.VCO_DIV 2
+ad_ip_parameter sys_audio_clkgen CONFIG.VCO_MUL 21
+ad_ip_parameter sys_audio_clkgen CONFIG.CLK0_DIV 85.5
+
+ad_connect sys_cpu_clk sys_audio_clkgen/clk
+ad_connect sys_i2s_mclk sys_audio_clkgen/clk_0
+
+# i2s ip
+ad_ip_instance axi_i2s_adi axi_i2s_adi
+ad_ip_parameter axi_i2s_adi CONFIG.DMA_TYPE 0
+ad_ip_parameter axi_i2s_adi CONFIG.S_AXI_ADDRESS_WIDTH 32
+
+# dma
+ad_ip_instance axi_dmac i2s_tx_dma
+ad_ip_parameter i2s_tx_dma CONFIG.DMA_TYPE_SRC 0
+ad_ip_parameter i2s_tx_dma CONFIG.DMA_TYPE_DEST 1
+ad_ip_parameter i2s_tx_dma CONFIG.CYCLIC 1
+ad_ip_parameter i2s_tx_dma CONFIG.AXI_SLICE_SRC 0
+ad_ip_parameter i2s_tx_dma CONFIG.AXI_SLICE_DEST 0
+ad_ip_parameter i2s_tx_dma CONFIG.ASYNC_CLK_DEST_REQ 0
+ad_ip_parameter i2s_tx_dma CONFIG.ASYNC_CLK_SRC_DEST 0
+ad_ip_parameter i2s_tx_dma CONFIG.ASYNC_CLK_REQ_SRC 0
+ad_ip_parameter i2s_tx_dma CONFIG.DMA_2D_TRANSFER 0
+ad_ip_parameter i2s_tx_dma CONFIG.DMA_DATA_WIDTH_DEST 32
+ad_ip_parameter i2s_tx_dma CONFIG.DMA_DATA_WIDTH_SRC 64
+
+ad_ip_instance axi_dmac i2s_rx_dma
+ad_ip_parameter i2s_rx_dma CONFIG.DMA_TYPE_SRC 1
+ad_ip_parameter i2s_rx_dma CONFIG.DMA_TYPE_DEST 0
+ad_ip_parameter i2s_rx_dma CONFIG.CYCLIC 1
```

Comment:
Do we need cyclic support for the RX DMA ?

Reply:
apparently we do